Source/FieldSolver/SpectralSolver/SpectralAlgorithms/AvgGalileanAlgorithm.cpp

```diff
@@ -0,0 +1,371 @@
+                    X4(i,j,k) = I*kv*X1(i,j,k) - theta*theta*S_ck(i,j,k)/ep0;
+                }
+                if ( nu == 0) {
+                    X1(i,j,k) = (1._rt - C(i,j,k)) / (ep0*c*c*k_norm*k_norm);
+                    X2(i,j,k) = (1._rt - S_ck(i,j,k)/dt) / (ep0*k_norm*k_norm);
+                    X3(i,j,k) = (C(i,j,k) - S_ck(i,j,k)/dt) / (ep0*k_norm*k_norm);
+                    X4(i,j,k) = -S_ck(i,j,k)/ep0;
+
+                    Psi1(i,j,k) = (-S1(i,j,k) + S3(i,j,k)) / (c*k_norm*dt);
+                    Psi2(i,j,k) = (-C1(i,j,k) + C3(i,j,k)) / (c2*k_norm*k_norm*dt);
+                    Psi3(i,j,k) = 1._rt;
+                    A1(i,j,k) = (c*k_norm*dt + S1(i,j,k) - S3(i,j,k)) / (c*c2 * k_norm*k_norm*k_norm * dt);
+                    A2(i,j,k) =  (c*k_norm*dt + S1(i,j,k) - S3(i,j,k)) / (c*c2 * k_norm*k_norm*k_norm * dt);
+                    CRhoold(i,j,k) = 2._rt * I * S1(i,j,k)  * ( dt*C(i,j,k) - S_ck(i,j,k))
+                                    / (c*k_norm*k_norm*k_norm*dt*dt*ep0);
+                    CRhonew(i,j,k) =  - I * (c2* k_norm*k_norm * dt*dt - C1(i,j,k) + C3(i,j,k))
+                                    / (c2 * k_norm*k_norm*k_norm*k_norm * ep0 * dt*dt);
+                    Jcoef(i,j,k) = (-C1(i,j,k) + C3(i,j,k)) / (c2*ep0*k_norm*k_norm*dt);
+                }
+                if ( nu == 1.) {
+                    X1(i,j,k) = (1._rt - e_theta*e_theta + 2._rt*I*c*k_norm*dt) / (4._rt*c*c*ep0*k_norm*k_norm);
+                    X2(i,j,k) = (3._rt - 4._rt*e_theta + e_theta*e_theta + 2._rt*I*c*k_norm*dt) / (4._rt*ep0*k_norm*k_norm*(1._rt- e_theta));
+                    X3(i,j,k) = (3._rt - 2._rt/e_theta - 2._rt*e_theta + e_theta*e_theta - 2._rt*I*c*k_norm*dt) / (4._rt*ep0*(e_theta - 1._rt)*k_norm*k_norm);
+                    X4(i,j,k) = I*(-1._rt + e_theta*e_theta + 2._rt*I*c*k_norm*dt) / (4._rt*ep0*c*k_norm);
+                }
+
+            } else { // Handle k_norm = 0, by using the analytical limit
+              C(i,j,k) = 1._rt;
+              S_ck(i,j,k) = dt;
+              C1(i,j,k) = 1._rt;
+              S1(i,j,k) =  0._rt;
+              C3(i,j,k) = 1._rt;
+              S3(i,j,k) = 0._rt;
+
+              X1(i,j,k) = dt*dt/(2._rt * ep0);
+              X2(i,j,k) = c2*dt*dt/(6._rt * ep0);
+              X3(i,j,k) = - c2*dt*dt/(3._rt * ep0);
+              X4(i,j,k) = -dt/ep0;
+              Theta2(i,j,k) = 1._rt;
+
+              Psi1(i,j,k) = 1._rt;
+              Psi2(i,j,k) = -dt;
+              Psi3(i,j,k) = 1._rt;
+              A1(i,j,k) = 13._rt * dt*dt /24._rt;
+              A2(i,j,k) = 13._rt * dt*dt /24._rt;
+              CRhoold(i,j,k) = -I*c2 * dt*dt / (3._rt * ep0);
+              CRhonew(i,j,k) = -5._rt*I*c2 * dt*dt / (24._rt * ep0);
+              Jcoef(i,j,k) = -dt/ep0;
+            }
+
+        });
+    }
+};
+
+/* Advance the E and B field in spectral space (stored in `f`)
+ * over one time step */
+void
+AvgGalileanAlgorithm::pushSpectralFields(SpectralFieldData& f) const{
```

Comment:
Here I think it would be better to add a space between the function name and the parentheses, following WarpX's style guidelines:
```suggestion
AvgGalileanAlgorithm::pushSpectralFields (SpectralFieldData& f) const{
```